index.js

```diff
@@ -696,6 +703,10 @@ class KeyringController extends EventEmitter {
     const { type, data } = serialized;
 
     const Keyring = this.getKeyringClassForType(type);
+    if (!Keyring) {
+      this.unsupported_keyrings.push(serialized);
+      return undefined;
```

Comment:
we can add the new return value to line 700 above. maybe

```
* @returns {Promise<Keyring>|undefined} The deserialized keyring or undefined if the keyring type is unsupported.
```

Reply:
Good catch! Slight correction, the type would be `Promise<Keyring|undefined>` since `undefined` here is still returned as a Promise.

Reply:
ah I missed the async. Thanks @Gudahtt!